waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -525,16 +525,23 @@
                        :service-id service-id}))))
   (let [rs-spec (replicaset-spec-builder-fn scheduler service-id service-description)
         request-namespace (k8s-object->namespace rs-spec)
-        request-url (str api-server-url "/apis/" replicaset-api-version "/namespaces/"
-                         request-namespace "/replicasets")
+        request-url (str api-server-url "/apis/" replicaset-api-version "/namespaces/" request-namespace "/replicasets")
         response-json (api-request request-url scheduler
                                    :body (utils/clj->json rs-spec)
                                    :request-method :post)
-        service (some-> response-json replicaset->Service)]
+        {:keys [k8s/replicaset-uid] :as service} (some-> response-json replicaset->Service)]
     (when-not service
       (throw (ex-info "failed to create service"
                       {:service-description service-description
                        :service-id service-id})))
+    (let [{:strs [min-instances]} service-description]
+      (when (> min-instances 1)
```

Comment:
I believe `pdb-spec-builder-fn` is treated as optional in the Scheduler constructor below, and I agree with that decision (i.e., not breaking current configs). Based on that, shouldn't we similarly have a non-nil check for `pdb-spec-builder-fn` in this conditional?

Reply:
Ah, I missed the `or` there. Sorry.

We need to have some way to turn this feature off. Either we have the default be `nil` (which means this whole block gets ignored), or we allow the user to specify a function that returns `nil`, and don't make the `api-request` below in that case.

I think we should definitely handle the second case as a way to disable this feature and make it more robust.

We might also want to change the default behavior back to no-op, but creating the PDBs is probably better.